MDANSE_GUI/Src/MDANSE_GUI/Tabs/Visualisers/MathRenderer.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+import base64
+import io
+import re
+
+from matplotlib import pyplot as plt
+
+
+class MathRenderer:
+    # Cache mapping the raw LaTex expression to its rendered image form
+    cache = {}
+
+    # Ignore the following expression
+    ignores = {r"\mathbf{q}": "q"}
+
+    def __init__(self, text: str) -> None:
+        self.raw_text = text
+
+    @staticmethod
+    def replace_ignored(text: str) -> str:
+        return MathRenderer.ignores[text]
+
+    @staticmethod
+    def ignore(text: str) -> bool:
+        return text in MathRenderer.ignores
+
+    @staticmethod
+    def containsMultiLineBlockExpression(text: str) -> bool:
+        return text in {".. math::", r"<br />.. math::<br />"}
+
+    @staticmethod
+    def containsBlockExpression(text: str) -> bool:
+        return text.startswith(".. math:")
+
+    @staticmethod
+    def containsInlineExpressions(text: str) -> bool:
+        pattern = r"(:math:`.*?`)"
+        substrings = re.split(pattern, text)
+        return len(substrings) > 1
+
+    @staticmethod
+    def processBlockExpression(text: str) -> list[tuple[str, bool]]:
+        return [(text[len(".. math:: ") :], True)]
+
+    @staticmethod
+    def processMultiLineBlockExpression(
+        strings: list[str], n: int
+    ) -> tuple[list[tuple[str, bool]], int]:
+        substrings = strings[n + 1 :]
+        group = []
+        for s in substrings:
+            if (not s) or (s == r"<br /><br />"):
+                break
+            group.append(s)
+
+        group = [s.replace(r"<br />", "") for s in group]
+        total = "".join(group)
+        if total.startswith(r"<br />") and total.endswith(r"<br />"):
+            result = total.split(r"<br />")[1]
+        else:
+            result = total.split(r"<br /><br />")[0]
+
+        return [(result, True)], n + len(group) + 1
```

Comment:
Again, you can use pattern matching here.

```suggestion
    def processMultiLineBlockExpression(
        self, strings: str
    ) -> None:
        pattern = r".. math::\s*\n(?:\s*\n)?(\s+)(.+)\n(\1.+\n)*"  # <- Should be class level variable
        
        for match in re.finditer(pattern, text):
            if match[3] is not None:
                body = "".join(more_itertools.prepend(match[2], match[3].splitlines()))
            else:
                body = match[2]
            text.replace(match[0], self.render(body))
```